src/common/com/intellij/plugins/haxe/ide/refactoring/introduce/HaxeIntroduceHandler.java

```diff
@@ -397,7 +418,8 @@ public PsiElement createDeclaration(HaxeIntroduceOperation operation) {
     final HaxeExpression initializer = operation.getInitializer();
     InitializerTextBuilder builder = new InitializerTextBuilder();
     initializer.accept(builder);
-    String assignmentText = "var " + operation.getName() + " = " + builder.result() + ";";
+    String suffix = builder.result() != null && builder.result().endsWith(";") ? "" : ";";
+    String assignmentText = "var " + operation.getName() + " = " + builder.result() + suffix;
```

Comment:
Here we are fixing whether or not we should place a semi-colon at the end of a statement.  In some of our PSI structures, the semi-colon is part of the extracted statement (e.g. `return blah;`).  In others, it is the following token and not part of the statement at all.